### ISSUE TITLE
prisma migrate dev --create-only asks for RESET DB

### ISSUE THREAD

**Issue by bandinopla** [prisma/prisma#14746]
### Bug description

i have an existing database. Created the schema for it with `prisma db pull`. But then when trying to create a migration baseline running prisma `migrate dev` it asks to reset the database... adding --create-only doesn't seem to work as espected.

### How to reproduce

have a simple database (mysql)
do `npx prisma db pull `
then `prisma migrate dev --name initial-migration --create-only `

### Expected behavior

since the schema was created from the already existing database, calling migrate dev with --create-only should create a migration.sql file with all the table creations necesary based on the schema. But instead it says it needs to wipe all data from the existing database...

### Prisma information

do `npx prisma db pull `
then `prisma migrate dev --name initial-migration --create-only `

### Environment & setup

- OS: win 10
- Database: MySQL
- Node.js version: v14.18.3

### Prisma Version

```
prisma                  : 4.2.0
@prisma/client          : 4.2.0
Current platform        : windows
Query Engine (Node-API) : libquery-engine 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt 2920a97877e12e055c1333079b8d19cee7f33826 (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : 2920a97877e12e055c1333079b8d19cee7f33826
Studio                  : 0.469.0
```

**Comment by janpio** [comment]
Are you following https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/baselining?

**Comment by bandinopla** [comment]
> Are you following https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/baselining?

i can't get to that point, to baseline you need to have the initial migration files created.
Im trying to create that initial file using a database i have in development that has data in it... i was asuming that calling migrate dev with --create-only would indeed create this migration.sql without touching the data, but it asks to do a hard reset for some reason.

I have read it has something to do with the shadow database being empty and the dev database being populated with tables... so they have diferences so it asks to do a reset... but i would espect the flag "--create-only" to be interpeted by prisma like "just do the sql and leave the databases alone"

**Comment by janpio** [comment]
So you are currently following https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#update-the-development-environment and getting stuck?

**Comment by bandinopla** [comment]
> These two assumptions are both not correct. Usage of `migrate dev` is unfortunately currently expected to ask for resets for various reasons. Optimally you can just turn your data into a seed script, and then resetting would not be a problem any more.

Yes but No. I didn't called `migrate dev` alone... i bringed our little friend called `--create-only` to the party. That flag should just see the diff between the shadow db and the one pointed in the `.env` do the SQL with the diffs and shut up...

in a way it feels like the current implementation opted to go for the lazy route... just do a reset if shadow and front db dont look the same... but it is not taking into account the flag and the implications of using it (the developer not wanting to do anything on the db)

if i didn't used the flag then yes, like you said. But using it, it should generate the .sql and end it there.

**Comment by janpio** [comment]
What I wrote applies to all parameters of `migrate dev`.

The important part of the question was here though:

> So you are currently following [prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#update-the-development-environment](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#update-the-development-environment) and getting stuck?

**Comment by bandinopla** [comment]
i read all the docs. What im saying is that it is not correct how it is behaving. The flag name is "create only", not "reset also"... the portion of code in your implementation responsable of resetting should run only if that flag doesn't exist and run once the developer removes the flag and re-runs the `prisma migrate dev`

Even in the docs it says:

> Run the following command to **create the first migration without applying it**, in case you need to **[modify the initial migration](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#work-around-features-not-supported-by-prisma-schema-language)**
`prisma migrate dev --name initial-migration --create-only`

So in that scenario, the developer want's to see the sql file just for review. Makes no sense to ask for a reset when the intention of the dev was to review BEFORE doing anything.

**Comment by janpio** [comment]
We are aware of that. See the other issues under the label: https://github.com/prisma/prisma/issues?q=is%3Aopen+label%3A%22topic%3A+prisma+migrate+dev+--create-only%22+sort%3Aupdated-desc I can not magically make it work different than how it currently works though.


I'll ask again:

> So you are currently following [prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#update-the-development-environment](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project#update-the-development-environment) and getting stuck?

**Comment by bandinopla** [comment]
yes i read that, i read all docs. 
The work arround im currently using is: 

1. run `npx prisma migrate db pull` on the existing db on my dev enviroment
2. switch the db on `.env` to point to an empty db
3. run `npx prisma migrate dev --create-only` and agree to "reset" -_-
4. switch the db on `.env` to point to the db i initially had 
5. "resolve" the initial migration created on step 3 `npx prisma migrate resolve --applied ...` 

and that solves the issue, then i continue normally...
